website-team/proposals/zola.md

```diff
@@ -0,0 +1,43 @@
+# Zolaification aka One Website To Rule Them All
+The current website is mainly divided between three separate components, the main website, the blog, and "Thanks". This has caused a few issues since their release as we have continued to work on it, as the styling has to be duplicated on each, and any change has to be replicated on each or the styling will be out of sync.
+
+Not only is styling not shared between websites but each component is written in a unique Rust app, the main website is dynamic web server hosted on Heroku while the blog and thanks are separate bespoke static site generators that are deployed on GitHub Pages.
+
+This makes the barrier to contributing quite high. It would significantly reduce the maintenance burden if we can consolidate these websites into a single website, and we could move away from requiring nightly rust for building and running the website. Ideally we would use a static site generator, as we haven't had a need for bespoke backend for any features on the website.
+
+## Why Zola?
+
+There are a lot of different static site generators, so there's a good reason ask why pick one over another. Zola has quite a few unique features that make it particularly appealing to us.
+
+
+- **Independently maintained. —** We're not the sole or the biggest user of Zola, so there is plenty of community interest in continuing to use and add to Zola.
+- **Tera templating language —** Tera is Zola's backing templating language that is a much more fully featured templating language that allows us to write a lot of the logic that we have in Rust directly in the Tera templates themselves.
+- **Written In Rust —** Since Zola is written in Rust, it's much easier for Rust Web Presence members to contribute features to Zola that they need for the website. In fact some of that is currently in progress to add features currently missing from Zola that is present in our bespoke apps such as fluent support and multiple RSS feeds.
+- **Asset collocation —** Zola by default allows you to host static content from the same directory as the source markdown, this removes some of the absolute URLs that required on the blog because the static assets directory is seperate.
+
+
+## URL Scheme Changes
+
+While Zola does provide the functionality we need it does not map directly to our current URL scheme, and merging the three websites will require moving the blog and thanks to be available under the `www.rust-lang.org` host. The blog currently has the following URL scheme.
```

Comment:
That's not true, we can have a single codebase but websites hosted on different domains. We simply need to generate the three domains in directories, and then serve each directory under its own domain name:

```
build/
├── blog
├── thanks
└── www
    └── assets
```

Reply:
That won't work because there will be shared assets such as fonts and CSS that located in subdirectories from the root of the build directory. Those assets will not be accessible through those domains without having to complicate the build process, and getting running it locally harder. With the goal of simplification I would not want building the website to be more complicated than running `zola serve` or `zola build`.

Reply:
I have looked a bit about Zola now, and with a slightly different structure I think it's possible to achive what I'm proposing.

![2020-06-01--18-40-30](https://user-images.githubusercontent.com/2299951/83431282-67367480-a437-11ea-99e3-612ba0354914.png)

This way one can run `zola serve` or `zola build` on the site they want to work on, and all the content we would share anyway is in the top-level directories, symlinked into each site.